src/rgw/rgw_rados.cc

```diff
@@ -6843,6 +6843,19 @@ int RGWRados::Object::Write::_do_write_meta(uint64_t size, uint64_t accounted_si
     op.setxattr(RGW_ATTR_MANIFEST, bl);
   }
 
+  /* get the storage class for bucket index update when is multipart upload */
+  if (meta.data && obj.in_extra_data) {
```

Comment:
this whole block should really be set from the caller. Can probably add `std::optional` storage_class member to MetaParams.